src/data_collector/polygon_news/storage.py

```
@@ -179,143 +146,179 @@
-            # Update keywords if different
             new_keywords = article_data.get("keywords", [])
-            if article.keywords != new_keywords:
-                article.keywords = new_keywords
+            if article.get("keywords") != new_keywords:
+                fields.append("keywords = %s")
+                params.append(new_keywords)
+
+            updated = False
+            if fields:
+                params.append(article_id)
+                sql = f"UPDATE polygon_news_articles SET {', '.join(fields)}, updated_at = now() WHERE id = %s"
```

Ruff (0.12.2)

Possible SQL injection vector through string-based query construction

(S608)